openpgp-dsm/src/lib.rs

```diff
@@ -885,69 +885,112 @@ pub fn generate_key(
     Ok(())
 }
 
+pub struct DsmKeyInfo {
+    name: String,
+    kid: Uuid,
+    hashing_algo: Option<HashAlgorithm>,
+    created_at: SdkmsTime,
+    last_used_at: SdkmsTime,
+    fingerprint: String,
+}
+
+impl From<&Sobject> for DsmKeyInfo {
+    /// Expects `DSM_LABEL_PGP` key to be present in metadata.
+    fn from(key: &Sobject) -> Self {
+        let key_md = KeyMetadata::from_sobject(&key).unwrap();
+        DsmKeyInfo { 
+            name: key.name.as_ref().unwrap().into(),
+            kid: key.kid.unwrap(),
+            hashing_algo: key_md.hash_algo,
+            created_at: key.created_at,
+            last_used_at: key.lastused_at,
+            fingerprint: key_md.fingerprint
+        }
+    }
+}
+
+impl DsmKeyInfo {
+    /// Prints key details in concise format, includes name, uuid, created_at
+     pub fn sobject_short_details_formatter(&self) -> String {
```

Comment:
Nit: When I first read the function name, I thought a formatter was to be returned.

It's a bit odd to have `let s: String = key_info.sobject_short_details_formatter();`. I think it is more readable to have `let `s: String = key_info.format_short();` or similar.